torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -251,3 +251,100 @@ def softmax_bwd_rule(op_schema: OpSchema) -> OutputSharding:
     ):
         raise RuntimeError("Cannot run _softmax_backward_data on sharding dimension!")
     return pointwise_rule(op_schema)
+
+
+def layer_norm_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
+    # args must be: input, normalized_shape, weight, bias, eps
+    assert len(op_schema.args_schema) == 5
+    (
+        input_schema,
+        normalized_shape,
+        weight_schema,
+        bias_schema,
+        _,
+    ) = op_schema.args_schema
+    # the current layer norm implementation requires that all input DTensor's sharding
+    # must be in form of OpStrategy
+    assert isinstance(input_schema, OpStrategy)
+    assert isinstance(normalized_shape, (int, Sequence, torch.Size))
+    normalized_size = (
+        [normalized_shape]
+        if isinstance(normalized_shape, int)
+        else list(normalized_shape)
+    )
+
+    input_shape = input_schema.output_shape
+    input_ndim = input_schema.output_ndim
+    axis = input_ndim - len(normalized_size)
+
+    # we use OpStrategy because the output (out, mean, rstd) should have the same placements
+    output_strategy = OpStrategy([])
+    for idx, input_arg_strategy in enumerate(input_schema.strategies):
+        input_args_specs = []
+        redistribute_costs = []
+        input_src_spec = input_arg_strategy.output_spec
+
+        # input tensor
+        input_target_spec = DTensorSpec(
+            mesh=mesh,
+            placements=_replicate_dims_start_at(input_src_spec.placements, axis),
+            tensor_meta=input_src_spec.tensor_meta,
+        )
+        input_args_specs.append(input_target_spec)
+        redistribute_costs.append(
+            generate_redistribute_costs(input_schema, input_target_spec)
+        )
+
+        if weight_schema is not None:
+            assert isinstance(weight_schema, OpStrategy)
+            weight_src_spec = weight_schema.strategies[idx].output_spec
+            weight_target_spec = DTensorSpec(
+                mesh=mesh,
+                placements=_replicate_dims_start_at(weight_src_spec.placements, 0),
```

Comment:
are we saying that we only support weight/bias replication for now? worth adding a comment

Reply:
Yes, to be more specific, we allow sharded weight/bias but we will replicate them before feeding them into the op.